src/index.ts

```diff
@@ -9,77 +9,169 @@ app.use(express.json());
+    });
 
     res.json(updatedPost);
   } catch (error) {
-    res.json({ error: `Post with ID ${id} does not exist in the database` });
+    res.status(500).json({ error: `Error updating post with ID ${id}` });
   }
 });
 
 app.delete(`/post/:id`, async (req, res) => {
   //delete the post
   const { id } = req.params;
-  const post = await prisma;
-  res.json(post);
+
+  try {
+    const post = await prisma.post.delete({
+      where: { id: Number(id) },
+    });
+
+    res.json(post);
+  } catch (error) {
+    res.status(500).json({ error: `Error deleting post with ID ${id}` });
+  }
 });
 
+
 app.get("/users", async (req, res) => {
   //return all the users
-  const users = await prisma;
-  res.json(users);
+  try {
+    const users = await prisma.user.findMany();
+    res.json(users);
+  } catch (error) {
+    res.status(500).json({ error: `Error fetching users` });
+  }
 });
 
 app.get("/user/:id/drafts", async (req, res) => {
   const { id } = req.params;
-  //return all posts where the published field equals false
-  const drafts = await prisma;
-  res.json(drafts);
+
+  try {
+    //return all posts where the published field equals false
+    const drafts = await prisma.post.findMany({
+      where: {
+        authorId: Number(id),
+        published: false,
+      },
+    });
+
+    res.json(drafts);
+  } catch (error) {
+    res.status(500).json({ error: `Error fetching drafts for user with ID ${id}` });
+  }
 });
 
+
 app.get(`/post/:id`, async (req, res) => {
   const { id }: { id?: string } = req.params;
-  //return the post
-  const post = await prisma;
-  res.json(post);
+
+  try {
+    //return the post
+    const post = await prisma.post.findUnique({
+      where: { id: Number(id) },
+    });
+
+    if (!post) {
+      return res.status(404).json({ error: `Post with ID ${id} does not exist in the database` });
+    }
+
+    res.json(post);
+  } catch (error) {
+    res.status(500).json({ error: `Error fetching post with ID ${id}` });
+  }
 });
 
 app.get("/feed", async (req, res) => {
   const { searchString, skip, take, orderBy } = req.query;
-  // 1. return all posts where the published field is set to true.
-  // 2. return the associated author with the post
-  // 3. skip the amount of posts specified
-  // 4. take the amount of posts specified
-  // 5. order the posts by the field `updated_at` descending or ascending basesd on the parameter `orderBy`
-  // 6. if the `searchString` parameter is not an empty, use the string to filter posts not matching the post titles or post content
 
-  const posts = await prisma.post.findMany();
+  try {
+    // Parse query parameters
+    const skipNumber = skip ? parseInt(skip as string) : undefined;
+    const takeNumber = take ? parseInt(take as string) : undefined;
+    const orderByObject = orderBy ? { updatedAt: orderBy as 'asc' | 'desc' } : undefined;
+
+    // Fetch posts
+    const posts = await prisma.post.findMany({
+      where: {
+        published: true,
+        OR: [
```

Comment:
If the `searchString` is empty we don't want to run this `OR` operator. Use a ternary to return an empty array instead.